include/inviwo/core/ports/portfactoryobject.h

```diff
@@ -94,8 +94,12 @@ class OutportFactoryObjectTemplate : public OutportFactoryObject {
 
     virtual std::unique_ptr<Outport> create() override { return std::make_unique<T>(className_); }
 
-    virtual std::unique_ptr<Outport> create(const std::string& identifier) override {
-        return std::make_unique<T>(identifier);
+    virtual std::unique_ptr<Outport> create(std::string_view identifier) override {
+        if constexpr (std::is_constructible_v<T, std::string_view>) {
+            return std::make_unique<T>(identifier);
+        } else {
+            return std::make_unique<T>(std::string(identifier));
```

Comment:
To harsh to force the ports to take a string_view and produce a nice interpretavle compile error instead?

Reply:
I have done that locally, but wanted to avoid braking code...